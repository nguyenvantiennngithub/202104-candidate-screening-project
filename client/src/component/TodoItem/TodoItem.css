.todo-container {
    display: flex;
    align-items: flex-start;
    border-top: 1px solid #363535;
    padding-bottom: 8px;
}
.todo-checkBox {
    color: #d8d8d8 !important;
}
.todo-content {
    margin-top: 12px;
    margin-bottom: 0;
    word-break: break-all;
}
.todo-container.completed .todo-content {
    color: #a18888 !important;
}

.completed {
    text-decoration: line-through;
}

.todo-date {
    color: #d8d8d8;
    font-size: 14px;
    margin: 8px 0 0 0;
}

.todo-date.dueDate {
    color: #d93025;
}
