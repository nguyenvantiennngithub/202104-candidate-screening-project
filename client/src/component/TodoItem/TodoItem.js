import React from "react";
import Checkbox from "@mui/material/Checkbox/";

import { useTodo } from "../../store/hooks";
import * as actionTodo from "../../store/todoAction";
import * as apiTodo from "../../api/todo";
import "./TodoItem.css";

const TodoItem = ({ data }) => {
    const { isCompleted, content, id, dueDate } = data;
    const date = new Date(dueDate);
    const now = new Date();
    const { dispatch } = useTodo();
    const isDueDate = date.getTime() - now.getTime() <= 0;

    const handleCheckComplete = () => {
        async function setComplete() {
            const payload = { isCompleted: !isCompleted, id };
            const { data } = await apiTodo.setCompleted(payload);

            if (data.status === "success")
                dispatch(actionTodo.setCompleted(payload));
        }

        setComplete();
    };
    return (
        <div
            className={
                isCompleted ? "todo-container completed" : "todo-container"
            }
        >
            <Checkbox
                color="success"
                className="todo-checkBox"
                onClick={handleCheckComplete}
                checked={isCompleted ? true : false}
            ></Checkbox>

            <div>
                <p className="todo-content">{content}</p>
                {dueDate && (
                    <p
                        className={
                            isDueDate ? "todo-date dueDate" : "todo-date"
                        }
                    >
                        {date.toLocaleString()}
                    </p>
                )}
            </div>
        </div>
    );
};

export default TodoItem;
