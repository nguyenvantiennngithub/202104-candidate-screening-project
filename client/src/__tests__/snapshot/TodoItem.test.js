import TodoItem from "../../component/TodoItem/TodoItem";
import Provider from "../../store/TodoProvider";
import renderer from "react-test-renderer";

describe("todo item", () => {
    it("props data normal case", () => {
        const todoItem = renderer.create(
            <Provider>
                <TodoItem
                    data={{
                        content: "Hello",
                        id: 1,
                        isCompleted: 0,
                        dueDate: "2022-05-13 08:52:00",
                    }}
                ></TodoItem>
            </Provider>
        );
        expect(todoItem.toJSON()).toMatchSnapshot();
    });
    // it("props data is empty", () => {
    //     const todoItem = renderer.create(
    //         <Provider>
    //             <TodoItem data={{}}></TodoItem>
    //         </Provider>
    //     );
    //     expect(todoItem.toJSON()).toMatchSnapshot();
    // });
    it("props data is undefined", () => {
        const todoItem = renderer.create(
            <Provider>
                <TodoItem></TodoItem>
            </Provider>
        );

        expect(todoItem.toJSON()).toMatchSnapshot();
    });
    it("props data.id is undefined", () => {
        const todoItem = renderer.create(
            <Provider>
                <TodoItem
                    data={{ id: undefined, content: "hello", isCompleted: 1 }}
                ></TodoItem>
            </Provider>
        );
        expect(todoItem.toJSON()).toMatchSnapshot();
    });
});
