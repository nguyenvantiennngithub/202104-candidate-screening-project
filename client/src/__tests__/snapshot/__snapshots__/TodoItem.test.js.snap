// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo item props data is undefined 1`] = `<div />`;

exports[`todo item props data normal case 1`] = `
<div
  className="todo-container"
>
  <span
    className="MuiCheckbox-root MuiCheckbox-colorSuccess MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSuccess PrivateSwitchBase-root todo-checkBox css-10drtbx-MuiButtonBase-root-MuiCheckbox-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={false}
      className="PrivateSwitchBase-input css-1m9pwf3"
      data-indeterminate={false}
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="CheckBoxOutlineBlankIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
  </span>
  <div>
    <p
      className="todo-content"
    >
      Hello
    </p>
    <p
      className="todo-date dueDate"
    >
      5/13/2022, 8:52:00 AM
    </p>
  </div>
</div>
`;

exports[`todo item props data.id is undefined 1`] = `<div />`;
